dkg/src/dkg_manager/mod.rs

```diff
@@ -1,30 +1,85 @@
 // Copyright © Aptos Foundation
-use crate::{dkg_manager::agg_trx_producer::AggTranscriptProducer, network::IncomingRpcRequest};
-use aptos_channels::aptos_channel;
+use crate::{agg_trx_producer::AggTranscriptProducer, network::IncomingRpcRequest, DKGMessage};
+use anyhow::{anyhow, bail, ensure, Result};
+use aptos_channels::{aptos_channel, message_queues::QueueStyle};
+use aptos_crypto::Uniform;
+use aptos_logger::error;
 use aptos_types::{
-    dkg::{DKGNode, DKGPrivateParamsProvider, DKGSessionState, DKGStartEvent, DKGTrait},
+    dkg::{
+        DKGNode, DKGPrivateParamsProvider, DKGSessionState, DKGStartEvent, DKGTrait,
+        DKGTranscriptMetadata,
+    },
     epoch_state::EpochState,
+    on_chain_config::ValidatorSet,
+    validator_txn::ValidatorTransaction,
 };
 use aptos_validator_transaction_pool as vtxn_pool;
 use futures_channel::oneshot;
-use futures_util::{FutureExt, StreamExt};
+use futures_util::{future::AbortHandle, FutureExt, StreamExt};
 use move_core_types::account_address::AccountAddress;
+use rand::thread_rng;
 use std::sync::Arc;
 
-pub mod agg_trx_producer;
+#[derive(Clone, Debug)]
+enum InnerState<DKG: DKGTrait> {
+    NotStarted,
+    InProgress {
+        start_time_us: u64,
+        public_params: DKG::PublicParams,
+        my_node: DKGNode,
```

Comment:
Maybe add a comment saying the partially aggregated trx is in the broadcast status? It is a bit confusing at first glance that InProgress state only has local trx.